<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="input_trusted_pose_with_cov_topic" default="/sensing/gnss/pose_with_covariance" description="Trusted pose input topic name. "/>
  <arg name="input_ndt_pose_with_cov_topic" default="/localization/pose_estimator/pose_with_covariance" description="NDT output pose input topic name. "/>
  <arg name="output_pose_with_covariance_topic" default="/localization/autoware_pose_covariance_modifier/pose_with_covariance" description="Estimated self position with covariance"/>
  <arg name="param_file" default="$(find-pkg-share autoware_pose_covariance_modifier_node)/config/autoware_pose_covariance_modifier.param.yaml"/>

  <node pkg="autoware_pose_covariance_modifier_node" exec="autoware_pose_covariance_modifier_node" name="autoware_pose_covariance_modifier_node" output="screen">
    <remap from="input_trusted_pose_with_cov_topic" to="$(var input_trusted_pose_with_cov_topic)"/>
    <remap from="input_ndt_pose_with_cov_topic" to="$(var input_ndt_pose_with_cov_topic)"/>
    <remap from="output_pose_with_covariance_topic" to="$(var output_pose_with_covariance_topic)"/>
    <param from="$(var param_file)"/>
  </node>
</launch>
